src/platforms/posix/px4_layer/px4_log.c

```diff
@@ -33,18 +54,44 @@ void px4_backtrace()
 
 __EXPORT void px4_log_modulename(int level, const char *moduleName, const char *fmt, ...)
 {
-	if (level <= __px4_log_level_current) {
-		PX4_LOG_COLOR_START
-		printf(__px4__log_level_fmt __px4__log_level_arg(level));
-		PX4_LOG_COLOR_MODULE
-		printf(__px4__log_modulename_pfmt, moduleName);
-		PX4_LOG_COLOR_MESSAGE
-		va_list argptr;
+	PX4_LOG_COLOR_START
+	printf(__px4__log_level_fmt __px4__log_level_arg(level));
+	PX4_LOG_COLOR_MODULE
+	printf(__px4__log_modulename_pfmt, moduleName);
+	PX4_LOG_COLOR_MESSAGE
+	va_list argptr;
+	va_start(argptr, fmt);
+	vprintf(fmt, argptr);
+	va_end(argptr);
+	PX4_LOG_COLOR_END
+	printf("\n");
+
+	/* publish an orb log message */
+	if (level >= _PX4_LOG_LEVEL_WARN && orb_log_message_pub) { //only publish important messages
+
+		struct log_message_s log_message;
+		const unsigned max_length = sizeof(log_message.text);
+		log_message.timestamp = hrt_absolute_time();
+
+		const uint8_t log_level_table[] = {
+			6, /* _PX4_LOG_LEVEL_ALWAYS */
+			7, /* _PX4_LOG_LEVEL_DEBUG */
+			4, /* _PX4_LOG_LEVEL_WARN */
+			3, /* _PX4_LOG_LEVEL_ERROR */
+			0  /* _PX4_LOG_LEVEL_PANIC */
```

Comment:
that's the alternative. my reasoning was that there are more levels in the log_message, which would lead to unused entries in the arrays (eg `__px4_log_level_str`).